app/src/main/java/org/connectbot/service/TerminalManager.java

```diff
@@ -711,7 +720,43 @@ private void reconnectPending() {
 				if (bridge == null) {
 					continue;
 				}
-				bridge.startConnection();
+
+				int sleepVal = 500;
+				int retryLimit = getRetries();
+				int retryCount = 0;
+
+				try {
+					Thread.sleep(sleepVal);
+				} catch(InterruptedException ex) {
+					Thread.currentThread().interrupt();
+				}
+
+				while ( ( retryLimit == 0 || retryCount <= retryLimit ) && bridge.isDisconnected() ){
+
+					bridge.startConnection();
+
+					if ( retryCount < 8 ) {
```

Comment:
Any kind of constants ("8" here and "2" below) should be extracted out to be in the class near the top.